server/lib/nl/fulfillment/triple.py

```diff
@@ -29,44 +33,37 @@
 _IN_TITLE = '{entity} is the {property} for:'
 
 
-# Gets the chart title for a list of entities, the property string,
-# and direction of the property
-def _get_title(entities, prop: str, out: bool) -> str:
-  title_format_str = _OUT_TITLE if out else _IN_TITLE
+# Gets the chart title for a list of entities and a property expression
+# TODO: revisit how titles should be displayed
+def _get_title(entities: List[Entity], prop_expression: str) -> str:
   entity_str = ', '.join([e.name or e.dcid for e in entities])
-  return title_format_str.format(property=prop, entity=entity_str)
+  if prop_expression in OUT_PROP_DISPLAY_NAME or prop_expression.startswith(
+      _OUT_ARROW):
+    prop_display_name = OUT_PROP_DISPLAY_NAME.get(
+        prop_expression, prop_expression[len(_OUT_ARROW):])
+    title_format_str = _OUT_TITLE
+  else:
+    prop_display_name = prop_expression[len(_IN_ARROW):]
+    title_format_str = _IN_TITLE
+  return title_format_str.format(property=prop_display_name, entity=entity_str)
 
 
-# Gets the chart vars for a property and direction if values exist for that
-# property and direction,
-def _get_chart_vars(uttr: nl_uttr.Utterance, prop: str, out: bool) -> ChartVars:
+# Adds chart for a property expression if values exist for that expression
+def _populate_prop_expression(uttr: nl_uttr.Utterance,
+                              prop_expression: str) -> bool:
   entity_dcids = [e.dcid for e in uttr.entities]
-  prop_with_direction = f'{_OUT_ARROW}{prop}' if out else f'{_IN_ARROW}{prop}'
   prop_values = get_property_value_from_expression(entity_dcids,
-                                                   prop_with_direction)
+                                                   prop_expression)
   if not any(prop_values.values()):
-    uttr.counters.err('triple_property_failed_existence', prop_with_direction)
-    return None
-  title = _get_title(uttr.entities, prop, out)
-  chart_vars = ChartVars(props=[prop_with_direction], title=title)
-  return chart_vars
-
-
-# Handles properties with no direction specified
-def _populate_any_direction_prop(uttr: nl_uttr.Utterance, prop: str):
-  # TODO: prefer index to use properties with arrows
-  out_chart_vars = _get_chart_vars(uttr, prop, True)
-  in_chart_vars = _get_chart_vars(uttr, prop, False)
-  chart_added = False
+    uttr.counters.err('triple_property_failed_existence', prop_expression)
+    return False
+  title = _get_title(uttr.entities, prop_expression)
+  cv = ChartVars(props=[prop_expression], title=title)
   state = PopulateState(uttr=uttr)
```

Comment:
I missed this before, why are we constructing `state` and not passing it to `populate` like others do?

Reply:
I followed how place overview and events does it (https://github.com/datacommonsorg/website/blob/d13b255a64c93427bd3aeb469a1067b0ec924528/server/lib/nl/fulfillment/fulfiller.py#L88-L93), should those actually also pass state instead of constructing state ..?

Reply:
good point.  I'd say pass state if you have it, since it gives you more info.  but leaving a deferring or leaving TODO is fine.

Reply:
updated to pass state here